js/adapt-assessmentArticleModel.js

```diff
@@ -24,809 +22,804 @@ define([
-      const allQuestionsAnswered = (numberOfQuestionsAnswered >= this._getCurrentQuestionComponents().length);
-      if (!allQuestionsAnswered) return;
-      this._onAssessmentComplete();
-    },
+    return questionModels;
+  },
 
-    _onAssessmentComplete: function() {
-      this.set('_attemptInProgress', false);
-      this._spendAttempt();
+  _overrideQuestionComponentSettings() {
+    const newSettings = this._getMarkingSettings();
 
-      var scoreAsPercent = this._getScoreAsPercent();
-      var score = this._getScore();
-      var maxScore = this._getMaxScore();
-      const minScore = this._getMinScore();
-      const correctCount = this._getCorrectCount();
-      const correctAsPercent = this._getCorrectAsPercent();
-      const questionCount = this._getQuestionCount();
+    // Add any additional setting overrides here
+    const questionConfig = this.getConfig()._questions;
+    if (questionConfig.hasOwnProperty('_canShowFeedback')) {
+      newSettings._canShowFeedback = questionConfig._canShowFeedback;
+    }
 
-      this.set({
-        _scoreAsPercent: scoreAsPercent,
-        _score: score,
-        _maxScore: maxScore,
-        _minScore: minScore,
-        _correctAsPercent: correctAsPercent,
-        _correctCount: correctCount,
-        _questionCount: questionCount,
-        _lastAttemptScoreAsPercent: scoreAsPercent,
-        _assessmentCompleteInSession: true,
-        _isAssessmentComplete: true
+    if (!_.isEmpty(newSettings)) {
+      const questionComponents = this._getAllQuestionComponents();
+      questionComponents.forEach(model => model.set(newSettings, { pluginName: '_assessment' }));
+    }
+  },
+
+  _setupQuestionListeners() {
+    this._removeQuestionListeners();
+    this.listenTo(this.getChildren(), 'change:_isInteractionComplete', this._onBlockCompleted);
+  },
+
+  _checkNumberOfQuestionsAnswered() {
+    const questionComponents = this._getCurrentQuestionComponents();
+    const numberOfQuestionsAnswered = questionComponents.filter(model => model.get('_isInteractionComplete')).length;
+    this.set('_numberOfQuestionsAnswered', numberOfQuestionsAnswered);
+  },
+
+  _removeQuestionListeners() {
+    this.stopListening(this.getChildren(), 'change:_isInteractionComplete', this._onBlockCompleted);
+  },
+
+  _onBlockCompleted(blockModel, value) {
+    if (value === false) return;
+    const questionModels = blockModel.findDescendantModels('question');
+    questionModels.forEach(questionModel => {
+      this._onQuestionCompleted(questionModel, value);
+    });
+  },
+
+  _onQuestionCompleted(questionModel, value) {
+    if (value === false) return;
+    if (!questionModel.get('_isInteractionComplete')) return;
+
+    let numberOfQuestionsAnswered = this.get('_numberOfQuestionsAnswered');
+    numberOfQuestionsAnswered++;
+    this.set('_numberOfQuestionsAnswered', numberOfQuestionsAnswered);
+
+    this._updateQuestionsState();
+    Assessment.saveState();
```

Comment:
Was this intentional? Are you sure it's still functioning?

Reply:
Ive reverted that in the new commit